models.js

```diff
@@ -53,12 +53,46 @@ function getCommentsByArticleIdData(id){
     WHERE article_id = $1
     ORDER BY created_at DESC;`, [id]).then((res) =>{
         if (res.rows.length === 0){
-            return Promise.reject({ error: 404, message: `Comments for Article with id ${id} does not exist.` });
+            return Promise.reject({ status: 404, msg: `Comments for Article with id ${id} does not exist.`});
         }
         return res.rows
     })
 
 }
 
+function insertCommentIntoDb(id, comment) {
+    const { username, body } = comment;
 
-module.exports = {getTopicData, getArticleIdData, getArticleData, getCommentsByArticleIdData}
\ No newline at end of file
+    if (!username || !body) {
+        return Promise.reject({ status: 400, msg: 'bad request' });
+    }
+
+    return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [id])
+      .then((articleRes) => {
+        if (articleRes.rows.length === 0) {
+          return Promise.reject({ status: 404, msg: 'Article not found.' });
+        }
+        return db.query(`SELECT * FROM users WHERE username = $1;`, [username]);
```

Comment:
There is no need for three database queries here. If the user does not exist in the database psql will reject for you due to a foreign key constraint in the seed. You do need to check the article exists, I would recommend refactoring to make use of a utility function for this.